### PR TITLE
Belen serer locc

### PR DESCRIPTION
**Pull request by belenserer**
I created a color theme, deleted some comments and empty components and now I am trying to do the PR from my repo

### CODE REVIEW

File: Challenge/src/components/ListOfCrypto.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import React, { useEffect, useState } from "react";
+import { FlatList, Text, View, Image, StyleSheet } from "react-native";
+import list from "./list";
+import styles from "./styles/stylesList";
+
+const ListOfCurrencies = () => {
+    return (
```
Comment (thom-as-thom) [inline Challenge/src/components/ListOfCrypto.tsx]: use implicit return

File: Challenge/src/components/ListOfCrypto.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        <View style={styles.container}>
```
Comment (thom-as-thom) [inline Challenge/src/components/ListOfCrypto.tsx]: It's a good idea to use safe area view.

File: Challenge/src/components/ListOfCrypto.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            {list.map(c => (
+                <>
+                    <View style={styles.crypto} key={c.id}>
+                        <View style={styles.crypoProfile}>
```
Comment (thom-as-thom) [inline Challenge/src/components/ListOfCrypto.tsx]: Crypto profile could be a component

**Review by thom-as-thom — CHANGES_REQUESTED** [review]

File: Challenge/src/components/ListOfCrypto.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import React, { useEffect, useState } from "react";
```
Comment (thom-as-thom) [inline Challenge/src/components/ListOfCrypto.tsx]: dete unused imports

File: Challenge/src/components/ListOfCrypto.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import { FlatList, Text, View, Image, StyleSheet } from "react-native";
+import list from "./list";
+import styles from "./styles/stylesList";
+
+const ListOfCurrencies = () => {
+    return (
+        <View style={styles.container}>
+            {list.map(c => (
```
Comment (tomasdematos) [inline Challenge/src/components/ListOfCrypto.tsx]: use FlatLIst,

File: App.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import CryptoList from './src/components/cryptoList';
+import {SafeAreaView} from 'react-native';
+
+
+
+
+const App = () => {
```
Comment (tomasdematos) [inline App.tsx]: use implicit return 
const App = () => ()

File: App.tsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  return (
+
```
Comment (tomasdematos) [inline App.tsx]: delete blank spaces

File: config/utils/themes.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
+const Blue = '#385775'
```
Comment (tomasdematos) [inline config/utils/themes.ts]: the format should be : 
```
const theme = {
  colors: {
     black: '#000000',
     .....
  }
}

export default theme;
```

and delete all the const

File: src/components/cryptoList.tsx [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+  return (
+    <>
+      <TopBar>
+         <Text>CryptoTracker Pro</Text>
+        <ProfilePhoto source={avatar} /> 
+      </TopBar> 
+      <FlatList
+        data={cryptos}
+        renderItem={renderItem}
+        keyExtractor={item => item.id}
```
Comment (tomasdematos) [inline src/components/cryptoList.tsx]: you can use destructuring; 

```
keyExtractor={{id] => id}
```

File: src/components/list.tsx [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+  ContainerRight,
+  ContainerChange,
+  Arrow,
+  TextPercent,
+  TextGreen,
+  TextRed,
+  Line,
+
+} from './styles/stylesList';
+import greenarrow from './img/greenarrow.png';
```
Comment (tomasdematos) [inline src/components/list.tsx]: use PascalCase: 
GreenArrow, RedArrow

File: src/components/list.tsx [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+          <TextCoin>{item.name}</TextCoin>
+          <TextSymbol>{item.symbol}</TextSymbol>
+        </ContainerText>
+      </ContainerLeft>
+      <ContainerRight>
+        <ContainerChange>
+          <TextCoin>${item.price}</TextCoin>
+
+          <TextPercent>
+            {item.percent > 0 ? (
```
Comment (tomasdematos) [inline src/components/list.tsx]: redundant validation: 
you shoud use one component for this:
```
                <Arrow source={item.percent > 0  ? greenarrow  : redarrow } />
                <TextPercentage positive={item.percent > 0}> {item.percent}% </TextPercentage>
 ```
 
 Styled-components can take props, so you can validate the color dynamically inside the style

File: src/components/list.tsx [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+                <Arrow source={redarrow} />
+                <TextRed> {item.percent}% </TextRed>
+              </>
+            )}
+          </TextPercent>
+
+        </ContainerChange>
+      </ContainerRight>
+    </Container>
+
```
Comment (tomasdematos) [inline src/components/list.tsx]: delete black spaces, only use one of is necessary

File: src/components/styles/stylesCryptoList.ts [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+  width: 55px;
+  height: 55px;
+  border-radius: 100px;
+`;
+
+export const Container = styled(ViewRN)`
+  margin-top: 48px;
+  display: flex;
+  align-items: center;
+
```
Comment (tomasdematos) [inline src/components/styles/stylesCryptoList.ts]: delete blank lines, and add a space between this and the next style

File: src/components/styles/stylesList.ts [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+  padding-right: 7.5%;
+  padding-left: 3%;
+  
+`;
+
+export const ContainerText = styled(ViewRN)`  
+  display: flex;
+  justify-content: center;
+  text-align: left;
+
```
Comment (tomasdematos) [inline src/components/styles/stylesList.ts]: delete blank line

File: src/components/styles/stylesList.ts [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+export const ContainerChange = styled(ViewRN)`
+  height: 100%;  
+  padding-horizontal: 10px;
+  justify-content: flex-end;
+  display: flex;
+  justify-content: flex-start;
+  align-items: space-between;
+  padding: 24px;
+`;
+export const TextPercent = styled(TextRN)`
```
Comment (tomasdematos) [inline src/components/styles/stylesList.ts]: add a empty space

File: src/components/styles/stylesList.ts [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+  flex-direction: row;
+  justify-content: flex-end;
+
```
Comment (tomasdematos) [inline src/components/styles/stylesList.ts]: delete this line

File: src/components/styles/stylesList.ts [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+  height: 2px;
+  width: 85%;
+  margin-vertical: 10px;
+  margin-horizontal: 24px;
+`;
+
+export const CryptoLogo = styled(ImageRN)`
+  width: 50px;
+  height: 50px;
+
```
Comment (tomasdematos) [inline src/components/styles/stylesList.ts]: delete this lines

File: src/components/styles/stylesList.ts [hunk 18]
```diff
@@ -0,0 +1,104 @@
...
+  width: 50px;
+  height: 50px;
+  
+
+`;
+
+export const Arrow = styled(ImageRN)`
+  width: 13px;
+  height: 13px;
+
```
Comment (tomasdematos) [inline src/components/styles/stylesList.ts]: delete blank line

File: src/components/cryptoList/index.tsx [hunk 19]
```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import {TouchableOpacity, FlatList} from 'react-native';
```
Comment (thom-as-thom) [inline src/components/cryptoList/index.tsx:2]: delete unnecessary imports

Comment (belenserer) [reply src/components/cryptoList/index.tsx:2]: React is unnecesary? I am using TouchableOpacity and FlatList

**Pull request merged**